ansible/roles/install-microshift/tasks/main.yml

```diff
@@ -0,0 +1,237 @@
+      ansible.builtin.command: subscription-manager repos --enable={{ item }}
+      with_items: "{{ enable_repos }}"
+  when: ansible_distribution == "RedHat"
+
+- name: CentOS tasks
+  block:
+    - name: install firewalld & packages
+      ansible.builtin.dnf:
+        name: "{{ install_packages}}"
+        state: present
+        update_cache: true
+
+    - name: install node-exporter
+      ansible.builtin.dnf:
+        name:
+        - golang-github-prometheus-node-exporter
+        state: present
+
+    - name: download cadvisor
+      ansible.builtin.get_url:
+        url: "{{ cadvisor_url }}"
+        dest: /usr/bin/cadvisor
+        checksum: "{{ cadvisor_checksum }}"
+        mode: '0755'
+
+    - name: install systemd unit for cadvisor
+      ansible.builtin.template:
+        src: cadvisor.service.j2
+        dest: /usr/lib/systemd/system/cadvisor.service
+        backup: true
+
+    - name: start and enable systemd services
+      ansible.builtin.systemd:
+        name: "{{ item }}"
+        daemon_reload: yes
+        state: started
+        enabled: yes
+      with_items: "{{ systemd_services }}"
+  when: ansible_distribution == "CentOS"
+
+- name: install kubernetes repo
+  ansible.builtin.template:
+    src: kubernetes.repo.j2
+    dest: /etc/yum.repos.d/kubernetes.repo
+
+- name: install kubectl
+  ansible.builtin.dnf:
+    name:
+    - kubectl
+    state: present
+
+- name: create .kube home dir
+  ansible.builtin.file:
+    path: /root/.kube/
+    state: directory
+
+- name: create symbolic link for kubeconfig
+  ansible.builtin.file:
+    src: "{{ remote_kubeconfig_path }}"
+    path: "/root/.kube/config"
+    state: link
+    force: yes
+
+- name: permit traffic in trusted zone from CIDR
+  ansible.posix.firewalld:
+    source: 10.42.0.0/16
+    state: enabled
+    immediate: yes
+    permanent: yes
+    zone: trusted
+
+- name: permit traffic in public zone for services
+  ansible.posix.firewalld:
+    service: "{{ item }}"
+    state: enabled
+    immediate: yes
+    permanent: yes
+    zone: public
+  with_items: "{{ firewall_services }}"
+
+- name: permit traffic in public zone for ports
+  ansible.posix.firewalld:
+    port: "{{ item }}"
+    state: enabled
+    immediate: yes
+    permanent: yes
+    zone: public
+  with_items: "{{ firewall_ports }}"
+
+- name: upgrade all packages
+  ansible.builtin.dnf:
+    name: "*"
+    state: latest
+  notify: reboot machine
+
+- name: flush handlers
+  ansible.builtin.meta: flush_handlers
+
+- name: record initial disk space
+  vars:
+    filename: disk0.txt
+  include_tasks: disk.yml
+  loop: "{{ du_dirs }}"
+
+- name: enable microshift copr repo
+  community.general.copr:
+    name: '@redhat-et/microshift'
+    state: enabled
+
+- name: enable cri-o module
+  ansible.builtin.command: dnf module enable -y cri-o
+
+- name: install microshift package
+  ansible.builtin.dnf:
+    name:
+    - microshift
+    state: present
+    update_cache: true
```

Comment:
I think we may want to either build RPMs locally, or download packages from brew.

- https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8.md#build-microshift
- https://github.com/openshift/microshift/blob/main/docs/rpm_packages.md

Reply:
That's an interesting problem as on one hand we're doing a build on super underpowered hardware (local on the microshift node). Alternatively brew is behind the VPN so not easy to pull to AWS or outside in general.

However we could either build the RPMs locally and then push them to the MicroShift node, or follow that same pattern to pull the RPMs from brew locally and then push them to the node.

Reply:
MicroShift binary by itself does not require much resources to be built, especially on development VMs.
Anyway, let's push this task to the second round of enhancements.

Reply:
I've added the RPM building on the MicroShift node, takes around 9-10m.